### ISSUE TITLE
ToMany Hydratation

### ISSUE THREAD

**Issue by Webini** [doctrine/DoctrineModule#527]
Hi, i would like to do this kind of hydratation
sample data:
``` 
{
    "toManyField": [
        { id: 45, name: "test" },
        { id: 46, name: "test46" }
    ]
}
```
in entity like
```
class TestEntity {
    /** @var Test **/    
    protected $toManyField;

    ...
}

class Test {
    /** identifier **/
    protected $id;
    /** @var string **/
    protected $name;
    ...
}
``` 

When i use DoctrineHydrator, i got an "Unrecognized identifier fields" exception.
In https://github.com/doctrine/DoctrineModule/blob/master/src/DoctrineModule/Stdlib/Hydrator/DoctrineObject.php#L420 the find method use my 'id' and 'name' field for searching existing entities. Is that a normal behavior ? How can we then update our entity (line 438 ?) if the code throw an exception in the find step ? Or can i define the identifier fields for the find ?

Thank
Nico.

**Comment by phpboyscout** [comment]
Hi Nico, 

I've created a PR that may solve your issue. I'm just waiting for it to be approved and merged https://github.com/doctrine/DoctrineModule/pull/535

**Comment by gianarb** [comment]
Resolved e8158536a4849f97160ff7220192f2852c507082
Thanks if the problem persist notify me please and I will open this pr :)
